integration_tests/broker_test.go

```diff
@@ -0,0 +1,199 @@
+package integration_test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+
+	"github.com/alphagov/paas-compose-broker/catalog"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	uuid "github.com/satori/go.uuid"
+)
+
+var (
+	responseRecorder  *httptest.ResponseRecorder
+	instanceID        string
+	serviceID         string
+	planID            string
+	bindingID         string
+	paramJSON         string
+	acceptsIncomplete bool
+)
+
+var _ = Describe("Broker", func() {
+
+	BeforeEach(func() {
+		responseRecorder = httptest.NewRecorder()
+		instanceID = uuid.NewV4().String()
+		serviceID = "Service-1"
+		planID = "Plan-1"
+		bindingID = uuid.NewV4().String()
+		paramJSON = "{}"
+		acceptsIncomplete = true
+	})
+
+	It("serves the catalog endpoint", func() {
+		req := newRequest("GET", "/v2/catalog", nil)
+		resp := serveRequest(responseRecorder, req)
+		Expect(responseRecorder.Code).To(Equal(http.StatusOK))
+		body := readResponseBody(resp)
+		returnedCatalog, err := catalog.New(bytes.NewBuffer(body))
```

Comment:
We decided to use new catalog struct in the tests that would consist of only the fields we need for assertions. We thought it might be cleaner than using `map[string]interface{}`. Also `listenPort` is now hardcoded in test so we can remove freeport dependency. This allowed us to remove brokerURL also.